src/io/syncfile.h

```diff
@@ -1,2 +1,39 @@
 MVMObject * MVM_file_open_fh(MVMThreadContext *tc, MVMString *filename, MVMString *mode);
 MVMObject * MVM_file_handle_from_fd(MVMThreadContext *tc, uv_file fd);
+
+/* Data that we keep for a file-based handle. */
+typedef struct {
+    /* File descriptor. */
+    int fd;
+
+    /* Is it seekable? */
+    short seekable;
+
+    /* Is it known to be writable? */
+    short known_writable;
+
+    /* Was the mode specified readonly? */
+    short mode_readonly;
```

Comment:
We're probably not going to have millions of files open at the same time and if we do, the size of this struct would be the least of our problems. But even so, previously using shorts for flags was no waste, as one int + 2 shorts = 64 bits which they would have taken anyway due to byte_position's alignment requirements. However now that we have more flags here, we use 64 more bits just to store 3 flags. I think it's time to move them into a single 32 bit flags field.